src/domain/products/components/customs-form/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react"
+import { Controller } from "react-hook-form"
+import InputField from "../../../../components/molecules/input"
+import Select from "../../../../components/molecules/select"
+import { Option } from "../../../../types/shared"
+import { countries } from "../../../../utils/countries"
+import FormValidator from "../../../../utils/form-validator"
+import { NestedForm } from "../../../../utils/nested-form"
+
+export type CustomsPayload = {
+  mid_code: string | null
+  hs_code: string | null
+  origin_country: Option | null
+}
+
+type CustomsFormProps = {
+  form: NestedForm<CustomsPayload>
+}
+
+/**
+ * Re-usable nested form used to submit customs information for products and their variants.
+ * @example
+ * <CustomsForm form={nestedForm(form, "customs")} />
+ */
+const CustomsForm = ({ form }: CustomsFormProps) => {
+  const {
+    register,
+    path,
+    control,
+    formState: { errors },
+  } = form
+
+  const countryOptions = countries.map((c) => ({
+    label: c.name,
+    value: c.alpha2,
+  }))
+
+  return (
+    <div className="grid grid-cols-2 gap-large">
+      <InputField
+        label="MID Code"
+        placeholder="XDSKLAD9999..."
+        {...register(path("mid_code"), {
+          pattern: FormValidator.whiteSpaceRule("MID Code"),
+        })}
+        errors={errors}
+      />
+      <InputField
+        label="HS Code"
+        placeholder="BDJSK39277W..."
+        {...register(path("hs_code"), {
+          pattern: FormValidator.whiteSpaceRule("HS Code"),
+        })}
+        errors={errors}
+      />
+      <Controller
+        name={path("origin_country")}
+        control={control}
+        render={({ field }) => {
+          return (
+            <Select
```

Comment:
todo: Select component is off in both its states:

**Closed**
![image](https://user-images.githubusercontent.com/59018053/186615465-35ac4178-b54d-4b89-99a6-1c1e337711bc.png)

**Open**
![image](https://user-images.githubusercontent.com/59018053/186615507-ad1bf4b1-5dc6-4469-9e01-ca3e4a6597af.png)

Reply:
So this happends with the current select component when you open it by clicking on the arrow button, and it is also the behaviour in the live version of admin. If you click on the actual input then it should open up the list. Will try to resolve some of these issues + the design of the select in a seperate PR.